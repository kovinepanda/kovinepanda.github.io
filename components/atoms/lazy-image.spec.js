import { mount } from "@vue/test-utils";
import LazyImage from './lazy-image';

describe('LazyImage', () => {
    it('renders correctly', () => {
        const blackPixel = 'data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs='
        const wrapper = mount(LazyImage, {
            propsData: {
                src: blackPixel,
                alt: 'black pixel'
            }
        })
        expect(wrapper.element).toMatchSnapshot()
      })
})
