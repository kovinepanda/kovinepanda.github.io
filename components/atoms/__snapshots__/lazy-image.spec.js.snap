// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImage renders correctly 1`] = `
<figure
  class="picture"
>
  <picture>
    <source
      data-srcset="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=?webp"
      type="image/webp"
    />
     
    <source
      data-srcset="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
      type="image/png"
    />
     
    <img
      alt="black pixel"
      class="lazyload"
      data-src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
      src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkqAcAAIUAgUW0RjgAAAAASUVORK5CYII="
    />
  </picture>
</figure>
`;
